modules/foreman_compute_resource.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) Philipp Joos 2017
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_compute_resource
+short_description: Manage Foreman Compute resources using Foreman API
+description:
+- Create and delete Foreman Compute Resources using Foreman API
+options:
+  name:
+    description: compute resource name
+    required: true
+  description: compute resource description
+    required: false
+  provider:
+    description: provider
+    required: false
+    default: None
+    choices: ["vmware", "libvirt", "ovirt"]
+  provider_auth:
+    description: provider authentication
+    required: false
+    default: None
+  locations: List of locations the compute resource should be assigned to
+    required: false
+    default: None
+  organizations: List of organizations the compute resource should be assigned to
+    required: false
+    default: None
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  verify_ssl:
+    description: verify ssl connection when communicating with foreman
+    required: false
+    default: true
+  state:
+    description: compute resource presence
+    required: false
+    default: present
+    choices: ["present", "absent", "latest"]
+notes:
+- Nailed down version of https://github.com/Nosmoht/python-foreman (@Nosmoht). Requires nailgun
```

Comment:
Huh? What's this for?

Reply:
A pun, i think => amusement
And a reference to the module where the functionality was found previosly.
Should it be removed?

Reply:
Oh, gotcha. It's fine and probably good to keep if we used any ideas from that module.